samples/python_interop/qiskit.ipynb

```diff
@@ -0,0 +1,482 @@
+      "metadata": {},
+      "outputs": [],
+      "source": [
+        "from qiskit import ClassicalRegister, QuantumRegister\n",
+        "from qiskit.circuit import (\n",
+        "    QuantumCircuit,\n",
+        ")\n",
+        "\n",
+        "from qsharp import QSharpError, TargetProfile\n",
+        "\n",
+        "qreg = QuantumRegister(3, name=\"q\")\n",
+        "creg = ClassicalRegister(3, name=\"c\")\n",
+        "qc = QuantumCircuit(qreg, creg)\n",
+        "qc.h([0, 1, 2])\n",
+        "qc.measure_all(add_bits=False)\n",
+        "\n",
+        "with qc.switch(creg) as case:\n",
+        "    with case(7):\n",
+        "        qc.x(0)\n",
+        "    with case(1, 2):\n",
+        "        qc.z(1)\n",
+        "    with case(case.DEFAULT):\n",
+        "        qc.cx(0, 1)\n",
+        "qc.measure_all(add_bits=False)\n",
+        "\n",
+        "backend = QSharpBackend()\n",
+        "\n",
+        "print(backend.qasm3(qc))"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "id": "fea9051a",
+      "metadata": {},
+      "source": [
+        "Using that same circuit, we can generate QIR which is used to run on quantum hardware."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "id": "0081f6f0",
+      "metadata": {},
+      "outputs": [],
+      "source": [
+        "backend = QSharpBackend(target_profile=TargetProfile.Adaptive_RI)\n",
+        "print(backend.qir(qc))"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "id": "97184460",
+      "metadata": {},
+      "source": [
+        "Not all programs can run on all hardware. Here we can try to target the `Base` profile, but we will get detailed errors on which parts of the program aren't supported."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "id": "06c31a67",
+      "metadata": {},
+      "outputs": [],
+      "source": [
+        "try:\n",
+        "    backend.qir(qc, target_profile=TargetProfile.Base)\n",
+        "except QSharpError as e:\n",
+        "    print(e)"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "id": "b06b7857",
+      "metadata": {},
+      "source": [
+        "## Errors\n",
+        "#### Unsupported language features, `QasmError`, and `QSharpError`\n",
+        "The modern QDK's interop with Qiskit is based on Qiskit's OpenQASM 3 support. Qiskit supports a subset of OpenQASM 3 features which may cause issues during conversion. If the Qiskit OpenQASM `Exporter` or OpenQASM parser don't support the feauture yet, a `QasmError` is raised prior to conversion. If there is a QASM parsing failure, this is likely an issue with the Qiskit libraries parsing or export functionality.\n",
+        "\n",
+        "Any failure to transform the OpenQASM into Q#'s internal representation will throw a `QasmError` to distinguish the scenarios. This is most likely due to a semantically invalid OpenQASM program as input.\n",
```

Comment:
Is this wording right? It states _"will throw a `QasmError` to distinguish the scenarios"_, but QasmError is the same error thrown in the preceding paragraph/problem, so it's not helping 'distinguish'.

Reply:
Looking, I did a lot of reorganization and might have messed up that part.